### PR TITLE
URA-872 - add end to end tests for 2 runs to 2 locations

### PR DESCRIPTION
**Pull request by jethror1**
- add end to end tests for 2 complete runs in 2 separate monitored directories uploading to 2 separate bucket locations
- rename `test_e2e.py` -> `test_one_complete_run_upload_to_one_remote_path.py`
- move test shared config to `__init__.py`
- separate test file creation to function `helper.create_files()`

```
jethro@jethro-T490:~/Projects/s3_upload/tests/e2e$ python3 -m pytest -v  .
=============================================================================================================================================== test session starts ================================================================================================================================================
platform linux -- Python 3.8.10, pytest-7.0.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.8.10', 'Platform': 'Linux-5.15.0-124-generic-x86_64-with-glibc2.29', 'Packages': {'pytest': '7.0.1', 'pluggy': '1.0.0'}, 'Plugins': {'Faker': '14.0.0', 'repeat': '0.9.3', 'subtests': '0.11.0', 'html': '4.1.0', 'metadata': '3.0.0', 'mock': '3.11.1', 'cov': '4.0.0', 'random-order': '1.1.1'}}
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
rootdir: /home/jethro/Projects/s3_upload/tests/e2e
plugins: Faker-14.0.0, repeat-0.9.3, subtests-0.11.0, html-4.1.0, metadata-3.0.0, mock-3.11.1, cov-4.0.0, random-order-1.1.1
collected 10 items                                                                                                                                                                                                                                                                                                 

test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_remote_files_upload_correctly PASSED                                                                                                                                                                                         [ 10%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_slack_post_message_after_uploading_as_expected PASSED                                                                                                                                                                        [ 20%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_upload_log_file_for_fully_uploaded_run_correct PASSED                                                                                                                                                                        [ 30%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_uploaded_run_not_picked_up_to_upload_again PASSED                                                                                                                                                                            [ 40%]
test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_when_nothing_to_upload_that_exit_code_is_zero PASSED                                                                                                                                                                         [ 50%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_remote_file_uploaded_correctly PASSED                                                                                                                                                            [ 60%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_slack_post_message_after_uploading_as_expected PASSED                                                                                                                                            [ 70%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_upload_log_files_for_fully_uploaded_run_1_correct PASSED                                                                                                                                         [ 80%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_uploaded_run_not_picked_up_to_upload_again PASSED                                                                                                                                                [ 90%]
test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_when_nothing_to_upload_that_exit_code_is_zero PASSED                                                                                                                                             [100%]

================================================================================================================================================ 10 passed in 4.25s ================================================================================================================================================
```

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/s3_upload/27)
<!-- Reviewable:end -->

### CODE REVIEW

File: tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        os.remove(os.path.join(TEST_DATA_DIR, "test_config.json"))
+
+        # delete the logger log files
+        for log_file in glob(os.path.join(TEST_DATA_DIR, "logs", "*log*")):
+            os.remove(log_file)
+
+        # clean up the remote files we just uploaded
+        bucket = boto3.resource("s3").Bucket(S3_BUCKET)
+        objects = bucket.objects.filter(
+            Prefix=cls.run_1_remote_path.replace("/sequencer_a", "")
```
Comment (fletchercr) [inline tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py:124]: Does this also need to be done for `cls.run_2_remote_path` with `sequencer_b`?

File: tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+        self.assertEqual(uploaded_files, expected_remote_files)
+
+    def test_upload_log_files_for_fully_uploaded_run_1_correct(self):
+        """
+        Test the upload logs for run_1 from sequencer_a are as expected
+        """
+        expected_top_level_log_contents = {
+            "run_id": "run_1",
+            "run path": self.run_1,
```
Comment (fletchercr) [inline tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py:169]: `run_path`? (in the single completed run too)

File: tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py (hunk 1 above)
Comment (fletchercr) [reply tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py:124]: Are tests for run_2 not required?

**Review by jethror1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/27)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @fletchercr)

___
*[`tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py` line 124 at r1](https://reviewable.io/reviews/eastgenomics/s3_upload/27#gh-1825768334:-OAbtyuQE4pcOXyP0CT8:b-po61un) ([raw file](https://github.com/eastgenomics/s3_upload/blob/d4e8d7cea89b2a3a4c4e08908cb59e50fc5bf059/tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py#L124)):*
<details><summary><i>Previously, fletchercr wrote…</i></summary><blockquote>

Are tests for run_2 not required? I guess the run_1 specific tests here would be the exact same for run_2
</blockquote></details>

yeah there should be I just missed adding them, have now added another test for run 2 since it's paths are different which is what I was trying to test
___
*[`tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py` line 169 at r1](https://reviewable.io/reviews/eastgenomics/s3_upload/27#gh-1825770664:-OAbu8H18DcQdoZBo3RP:b-inyb0a) ([raw file](https://github.com/eastgenomics/s3_upload/blob/d4e8d7cea89b2a3a4c4e08908cb59e50fc5bf059/tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py#L169)):*
<details><summary><i>Previously, fletchercr wrote…</i></summary><blockquote>

`run_path`? (in the single completed run too)
</blockquote></details>

turns out this was a typo in `io.py` when writing the log, have fixed it there and in the unit tests and here

**Comment by fletchercr** [comment]
*[`tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py` line 124 at r1](https://reviewable.io/reviews/eastgenomics/s3_upload/27#gh-1825768334:-OAc-N8b3C68r6iawhvP:bk3y3gi) ([raw file](https://github.com/eastgenomics/s3_upload/blob/d4e8d7cea89b2a3a4c4e08908cb59e50fc5bf059/tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py#L124)):*
<details><summary><i>Previously, jethror1 (Jethro Rainford) wrote…</i></summary><blockquote>

yeah there should be I just missed adding them, have now added another test for run 2 since it's paths are different which is what I was trying to test
</blockquote></details>

Do the objects in `cls.run_2_remote_path` need to be removed here too?

**Review by jethror1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/27)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @fletchercr)

___
*[`tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py` line 124 at r1](https://reviewable.io/reviews/eastgenomics/s3_upload/27#gh-1825768334:-OAc36LgB-0DCQe2Mwgg:b-swoo38) ([raw file](https://github.com/eastgenomics/s3_upload/blob/d4e8d7cea89b2a3a4c4e08908cb59e50fc5bf059/tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py#L124)):*
<details><summary><i>Previously, fletchercr wrote…</i></summary><blockquote>

Do the objects in `cls.run_2_remote_path` need to be removed here too?
</blockquote></details>

have now changed this to remove `cls.parent_remote_path` which is the parent of both, so both get removed together

**Review by fletchercr — APPROVED** [review]

**Review by fletchercr — APPROVED** [review]
> Reviewed 3 of 5 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/27)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @jethror1)

**Pull request merged**